src/styles/calendar.scss

```diff
@@ -0,0 +1,44 @@
+.calendar {
+  margin: 100px auto;
+  padding: $calendar-padding;
+  display: flex;
+  flex-wrap: wrap;
+  gap: $gap-size;
+  width: $calendar-width;
+
+  @each $number, $day in $week {
+    &--start-day--#{$day} :first-child {
+      margin-left: ($square-height + $gap-size) * ($number - 1);
+    }
+  }
+
+  @for $i from 28 through 31 {
+    &--month-length--#{$i} :nth-child(n + #{$i + 1}) {
+      display: none;
+    }
+  }
+
+  &__day {
+    height: $square-height;
+    width: $square-height;
+    border: 1px solid #000;
+    background-color: #eee;
+    text-align: center;
+    line-height: 100px;
+    transition: transform 0.5s;
+    cursor: pointer;
+
+    &:hover {
+      background-color: pink;
+      transform: translateY(-20px);
+    }
+
+    @for $day from 1 through 31 {
+      &:nth-child(n + #{$day})::after {
+        content: "#{$day}";
+        font-family: Arial, Helvetica, sans-serif;
```

Comment:
I think here you can leave only `content: "#{$day}"`. `font-family` and `font-size` can be moved above to `&__day` class don't need to add it here for each element in loop :)